indico/modules/events/abstracts/legacy.py

```diff
@@ -0,0 +1,506 @@
+from indico.modules.events.contributions.models.fields import ContributionFieldValue
+from indico.modules.events.contributions.models.persons import AuthorType, ContributionPersonLink
+from indico.modules.users.models.users import User, UserTitle
+from indico.util.string import to_unicode
+
+
+FIELD_TYPE_MAP = {
+    'input': 'text',
+    'textarea': 'text',
+    'selection': 'single_choice'
+}
+
+
+@signals.event.contributions.contribution_deleted.connect
+def _contribution_deleted(contrib, parent=None):
+    from MaKaC.review import AbstractStatusAccepted, AbstractStatusSubmitted
+    abstract = contrib.abstract
+    if abstract:
+        status = abstract.as_legacy.getCurrentStatus()
+        if isinstance(status, AbstractStatusAccepted):
+            if status.getTrack() is not None:
+                abstract.as_legacy.addTrack(status.getTrack())
+            abstract.as_legacy.setCurrentStatus(AbstractStatusSubmitted(abstract.as_legacy))
+            abstract.contribution = None
+
+
+def person_from_data(person_data, event):
+    user = User.find_first(~User.is_deleted, User.all_emails.contains(person_data['email'].lower()))
+    if user:
+        return EventPerson.for_user(user, event)
+
+    person = EventPerson.find_first(email=person_data['email'].lower())
+    if not person:
+        person = EventPerson(event_new=event, **person_data)
+    return person
+
+
+@no_autoflush
+def contribution_from_abstract(abstract, session):
+    from MaKaC.review import AbstractStatusAccepted
+
+    event = abstract.as_new.event_new
+    duration = session.default_contribution_duration if session else timedelta(minutes=15)
+    links = []
+
+    for auth in abstract.getAuthorList():
+        author_type = AuthorType.primary if abstract.isPrimaryAuthor(auth) else AuthorType.secondary
+        person_data = {
+            'title': UserTitle.from_legacy(to_unicode(auth.getTitle())),
+            'email': to_unicode(auth.getEmail()),
+            'address': to_unicode(auth.getAddress()),
+            'affiliation': to_unicode(auth.getAffiliation()),
+            'first_name': to_unicode(auth.getFirstName()),
+            'last_name': to_unicode(auth.getSurName()),
+            'phone': to_unicode(auth.getTelephone())
+        }
+
+        person = person_from_data(person_data, event)
+        person_data.pop('email')
+        links.append(ContributionPersonLink(author_type=author_type, person=person,
+                                            is_speaker=abstract.isSpeaker(auth), **person_data))
+
+    contrib = create_contribution(event, {'title': abstract.getTitle(), 'duration': duration,
+                                          'person_link_data': {link: True for link in links}})
+    contrib.abstract = abstract.as_new
+
+    contrib.description = abstract.as_new.description
+    for field_val in abstract.as_new.field_values:
+        new_val = ContributionFieldValue(contribution_field=field_val.contribution_field, data=field_val.data)
```

Comment:
check the last argument of `create_contribution()` (`custom_fields_data`); you can just pass the data there (`{'custom_ID': value}` dict)